core/api/storage_miner/storage_api.cpp

```diff
@@ -86,11 +86,60 @@ namespace fc::api {
       return result;
     };
 
-    api->SectorsStatus = [=](SectorNumber id,
-                             bool) -> outcome::result<ApiSectorInfo> {
-      // TODO(ortyomka): [FIL-421] implement it
+    api->SectorsStatus =
+        [=](SectorNumber id,
+            bool show_onchain_info) -> outcome::result<ApiSectorInfo> {
       OUTCOME_TRY(sector_info, miner->getSealing()->getSectorInfo(id));
-      return ApiSectorInfo{.state = sector_info->state};
+
+      std::vector<Piece> pieces{sector_info->pieces.size()};
+      std::vector<DealId> deals{sector_info->pieces.size()};
```

Comment:
https://en.cppreference.com/w/cpp/language/list_initialization
```suggestion
      std::vector<Piece> pieces(sector_info->pieces.size());
      std::vector<DealId> deals(sector_info->pieces.size());
```